### ISSUE TITLE
Submitting questions on firefox leads to the page reloading

### ISSUE THREAD

**Issue by Maximilian-Staab** [SecureAI-Tools/SecureAI-Tools#71]
# Issue

I can't send follow-up prompts when using Firefox.

## Steps to reproduce: 

1. Select "New Chat"
2. Enter a random prompt
3. Enter another   prompt into the "Say something..." box
4. Press enter to submit
5. The page reloads, the submitted prompt is deleted/not  worked on.

## Expected behavior:  

The new prompt is send to the model.

**Comment by JayNakrani** [comment]
Thanks for the detailed repro instructions. I just tested this on my Mac and Ubuntu Firefox and couldn't reproduce it unfortunately. Based on the description, looks like it's not preventing default form submission in your setup.

Quick Google searches tell me that we need to have the following to prevent default form submission on Firefox:
1. `cancelable: true` in the synthetic event ([ref](https://stackoverflow.com/a/56695472)) -- [we have this](https://github.com/SecureAI-Tools/SecureAI-Tools/blob/574d3733e8759925c02d952e6f70265a65c3d946/apps/web/lib/fe/components/chat.tsx#L435)
2. Call `preventDefault()` on form submission event -- [we have this too](https://github.com/SecureAI-Tools/SecureAI-Tools/blob/574d3733e8759925c02d952e6f70265a65c3d946/apps/web/lib/fe/components/chat.tsx#L404)
3. Return false from the form's `onSubmit` -- this is missing right now.

As I mentioned, I am not able to reproduce it, so can you test out potential fix? Here are two ways you can test it:
1. Either, run code from `jay/issue-71-potential-fix` branch (#72). You will need to run the web server from that branch and set-up development env to do that. OR
2. Save the following HTML code as a file, and open the file in Firefox. Then test it with and without that `return false` in it

```html
<html>
  <head>
    <script type="text/javascript">
      document.addEventListener("DOMContentLoaded", function(event) {
        const form = document.querySelector('#testform');
        const button = form.querySelector('#btn');

        form.onsubmit = function (e) {
          e.preventDefault();

          console.log('submitting the form ...');
          
          // TODO: Test the following two scenarios:
          //   1. With `return false` here
          //   2. Without the `return false` here
          return false;
        };

        const submitEvent = new Event('submit', { 
          cancelable: true
        });

        button.onclick = function () {
          console.log('button clicked');
          const cancelled = form.dispatchEvent(submitEvent);
          console.log('is cancelled', !cancelled);
        }
      });
    </script>
  </head>
  <body>
    <form id="testform">
      <input type="hidden" name="a" value="b">
      <button type="button" id="btn">click me!</button>
    </form>
  </body>
</html>
```

**Comment by Maximilian-Staab** [comment]
Hi Jay,

I've just updated my images (`public.ecr.aws/d8f2p0h3/secure-ai-tools:latest`) and can no longer reproduce this issue. It could be that I wasn't on `v0.0.3` previously and the changes from #52 fixed my issue as well. 

Thank you for the quick response and best wishes for the New Year!
